src/library/components/resource-lightbox.js

```diff
@@ -31,18 +38,29 @@ var ResourceLightbox = Component({
     portalObjectHelpers.processResource(resource)
 
     this.titleSuffix = document.title.split('|')[1] || ''
+    this.setState({
+      openAssign: false
+    })
     this.replaceResource(resource)
   },
 
   componentDidMount: function () {
+    if (this.state.openAssign) {
+      jQuery('#assign-button')[0].click()
+    }
     jQuery('.portal-pages-resource-lightbox-background, .portal-pages-resource-lightbox-container').fadeIn()
   },
 
   componentWillUnmount: function () {
-    document.title = this.props.savedTitle
+    document.title = this.state.savedTitle
     try {
-      window.history.replaceState({}, document.title, this.props.savedUrl)
+      if (this.state.parentPage !== '/' && this.state.parentPage !== PortalPages.initialPath) {
```

Comment:
Another comment here:
> When the parentPage is not `/` and it doesn't match the initiaPath then reload the page
> This can happen when a resource URL is opened directly. The parentPage URL parameter can be set to be a collection page but the initialPath will be the direct resource URL. 
> If the lightbox was opened from a collection page, the initialPath will be the collection page and the parentPage will be the collection page, so then we can just update the URL and close the lightbox.